backend/src/controllers/userProfile.ts

```diff
@@ -0,0 +1,139 @@
+import { Request, Response } from "express";
+import { z } from "zod";
+import { StatusCodes } from "http-status-codes";
+import { UserProfile } from "../models/userProfile";
+import { BaseError } from "sequelize";
+
+type ControllerItem = (req: Request, res: Response) => Promise<Response>;
+
+const controllerWrapper =
+	(controllerItem: ControllerItem) => async (req: Request, res: Response) => {
+		try {
+			return await controllerItem(req, res);
+		} catch (err) {
+			if (err instanceof z.ZodError) {
```

Comment:
reqSchema.parse(req) throws a ZodError, and we send the validation result back to frontend so they know what was wrong.